src/ripple/app/misc/FeeVoteImpl.cpp

```diff
@@ -171,7 +171,7 @@ FeeVoteImpl::doVoting(
             if (val->isFieldPresent (sfBaseFee))
             {
                 auto const vote = val->getFieldU64 (sfBaseFee);
-                if (isLegalAmount(vote))
+                if (vote <= std::numeric_limits<XRPAmount::value_type>::max())
```

Comment:
I _think_ I prefer the lower amount restriction in `isLegalAmount` to the larger amount in `value_type`. Is there a motivation for changing this that I'm missing?

Reply:
Yes. `vote` is a uint64. I got rid of the `isLegalAmount` override that explicitly takes a `uint64`, so the only remaining definition takes an `XRPAmount`. `XRPAmount` has an underlying int64 value is implicitly constructed from any integral value. If I call `isLegalAmount(vote)`, and vote is larger than the int64 limit, it will map to a negative value, and thus `isLegalAmount` will return true, which is a logic error.

I'll change it to check both.